Source/WebCore/page/Page.cpp

```diff
@@ -1746,6 +1746,14 @@ void Page::updateScreenSupportedContentsFormats()
     m_screenSupportsHDR = supportsHighDynamicRange;
     for (auto& rootFrame : m_rootFrames)
         rootFrame->screenSupportedContentsFormatsChanged();
+
+    forEachDocument([&] (Document& document) {
+        if (!document.hasHDRContent())
+            return;
+
+        if (RefPtr view = document.view())
+            view->setDescendantsNeedUpdateBackingAndHierarchyTraversal();
```

Comment:
If this going to traverse all the layers in the view, do we still need to call above
```
for (auto& rootFrame : m_rootFrames)
    rootFrame->screenSupportedContentsFormatsChanged();
```
`GraphicsLayerCA::setDrawsHDRContent()` will trigger `DrawsHDRContentChanged` if `drawsHDRContent` changes.

Reply:
Indeed! Tested and confirmed that it works.

Will update with the old code removed.